src/Digdir.Domain.Dialogporten.Application/Common/Extensions/ClaimsPrincipalExtensions.cs

```diff
@@ -120,22 +192,56 @@ public static bool TryGetAuthenticationLevel(this ClaimsPrincipal claimsPrincipa
             c.Type.StartsWith(AltinnClaimPrefix, StringComparison.Ordinal)
         ).OrderBy(c => c.Type);
 
-    private static bool TryGetOrgShortName(this ClaimsPrincipal claimsPrincipal, [NotNullWhen(true)] out string? orgShortName)
-        => claimsPrincipal.FindFirst(OrgClaim).TryGetOrgShortName(out orgShortName);
+    public static (UserIdType, string externalId) GetUserType(this ClaimsPrincipal claimsPrincipal)
+    {
+        if (claimsPrincipal.TryGetPid(out var externalId))
+        {
+            return (claimsPrincipal.HasScope(ServiceProviderScope)
+                ? UserIdType.ServiceOwnerOnBehalfOfPerson
+                : UserIdType.Person, externalId);
```

Comment:
This doesn't work. There is no `pid` claim, as the pid is supplied via a query parameter and is not part of the ClaimsPrincipal at all. 

One way to fix this might be to register another middleware that runs before this, which sole purpose is to check, if the request is authenticated, if 1) the service provider scope is provided and 2) the "EndUserId" query parameter is supplied (and perhaps syntactically valid). If so, it should add that value as a `Pid` claim to the ClaimsPrincipal. This should cause the `GetUserType` to work as intended (the logic here should be commented though).

Reply:
Added TODO comment for #386